custom_components/unifiprotect/sensor.py

```diff
@@ -142,40 +159,18 @@ def native_value(self):
             return self._device_data["motion_mode"]
 
         if self._device_type == DEVICE_TYPE_SENSOR:
-            if self._device_class == DEVICE_CLASS_BATTERY:
+            if self.entity_description.device_class == DEVICE_CLASS_BATTERY:
                 return self._device_data["battery_status"]
```

Comment:
Maybe add `ufp_data_key` in the EntityDescription so you don't need the big if block? and can do self._device_data[self.entity_description.ufp_data_key]

Reply:
Just made an edit to the sensor file. I could remove the whole if block. These EntityDescriptions are really useful

Reply:
Nice!

Reply:
I managed to optimize on sensor, number and switch. And I can't really see a smart way to do it on select.
I will now look at the binary_sensor file, to implement the suggestions.